presto-main/src/main/java/io/prestosql/server/security/UserExtractionUtil.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.server.security;
+
+import java.security.Principal;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.google.common.base.Strings.emptyToNull;
+
+final class UserExtractionUtil
+{
+    private UserExtractionUtil() {}
+
+    public static Optional<String> extractUser(Principal principal, Optional<Pattern> extractionPattern)
```

Comment:
the method should take `Pattern extractionPattern` and should be applied conditionally:

`extractionPattern .map(pattern -> extractUser( principal, pattern))`